### PR TITLE
Covert password_test.go from gocheck to std test

### PR DESCRIPTION
**Pull request by lxea**
While attempting to resolve flakiness in TestTiming I converted this to a test using standard go test utilities instead of gocheck. 

This PR does not resolve the flakiness yet however, but at least it can fail in parallel I guess?

### CODE REVIEW

File: lib/auth/password_test.go [hunk 1]
```diff
@@ -635,45 +640,45 @@ func (s *PasswordSuite) TestChangeUserAuthenticationWithErrors(c *C) {
 		// set new auth preference settings
 		authPreference.SetSecondFactor(tc.secondFactor)
 		err = s.a.SetAuthPreference(ctx, authPreference)
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 
 		_, err = s.a.changeUserAuthentication(ctx, tc.req)
-		c.Assert(err, NotNil, Commentf("test case %q", tc.desc))
+		require.Error(t, err, fmt.Sprintf("test case %q", tc.desc))
```
Comment (gabrielcorado) [inline lib/auth/password_test.go:646]:
```suggestion
		require.Errorf(t, err, "test case %q", tc.desc)
```

File: lib/auth/password_test.go [hunk 2]
```diff
@@ -635,45 +640,45 @@ func (s *PasswordSuite) TestChangeUserAuthenticationWithErrors(c *C) {
...
-		c.Assert(err, NotNil)
+		require.Error(t, err)
 		loginAttempts, _ = s.a.GetUserLoginAttempts(req.User)
-		c.Assert(len(loginAttempts), Equals, i+1)
+		require.Len(t, loginAttempts, i+1)
 	}
 
 	err := s.a.ChangePassword(req)
-	c.Assert(trace.IsAccessDenied(err), Equals, true)
+	require.Equal(t, true, trace.IsAccessDenied(err))
```
Comment (gabrielcorado) [inline lib/auth/password_test.go:678]:
```suggestion
	require.True(t, trace.IsAccessDenied(err))
```

**Review by gabrielcorado — APPROVED** [review]
LGTM. Just added a few code suggestions.

**Review by zmb3 — APPROVED** [review]

**Comment by zmb3** [comment]
FWIW, I looked at the flakiness aspect of `TestTiming` and I think it's just too timing dependent to be of value to us as is, and either needs to be removed or rewritten.

The idea here is that if a user doesn't exist, then by default the call would complete *faster* since it doesn't have to do any comparison. We mitigate this by intentionally slowing down the "user doesn't exist" case by doing a comparison against a dummy hash instead. This way we aren't leaking any information to an attacker about what users may or may not exist in the DB.

The irony here is that the user doesn't exist case is actually *slower* than the case where the user does exist, and by slowing it down further we're only increasing the gap.

I suspect this is because the amount of time it takes to compare the password hash is dwarfed by the extra `trace.IsNotFound` handling for the case when the user doesn't exist.
(In the doesn't exist case we're doing formatted logging that we wouldn't otherwise do, checking `trace.IsNotFound` _three times_, and generating backtraces for the errors which is a costly operation.

**Pull request merged**